source/dotnet/Library/AdaptiveCards/AdaptiveCardConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Reflection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveCardConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool CanWrite => false;
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var jObject = JObject.Load(reader);
+
+            var card = (AdaptiveCard) Activator.CreateInstance(objectType);
+            // Get the version from the payload
+            card.Version = null;
+
+            serializer.Populate(jObject.CreateReader(), card);
+
+            // TODO: return parse result error that version must be specified
+            if (card.Version == null)
+                return null;
+
+            // When objects without a valid "type" are parsed they return null but still get added to the collections, this removes them
+            card.Actions.RemoveAll(item => item == null);
+            card.Body.RemoveAll(item => item == null);
```

Comment:
Don't we need this logic on Container and Column too? So they would also need a custom converter?